src/traces/bar/sieve.js

```diff
@@ -40,14 +40,17 @@ function Sieve(traces, opts) {
     }
     this.positions = positions;
 
-    var dv = distinctVals(positions, {
-        unitMinDiff: opts.unitMinDiff
-    });
+    var dv = distinctVals(positions);
 
     this.distinctPositions = dv.vals;
     if(dv.vals.length === 1 && width1 !== Infinity) this.minDiff = width1;
     else this.minDiff = Math.min(dv.minDiff, width1);
 
+    var type = (opts.posAxis || {}).type;
+    if(type === 'category' || type === 'multicategory') {
+        this.minDiff = Math.min(1, this.minDiff);
```

Comment:
This looks like a great solution, and simpler than the previous logic. My only question is whether there's actually a benefit of doing `Math.min` vs just `minDiff = 1` in the (multi)category case? There's no way to specify input data that will result in non-integer positions here, is there?

Reply:
Good call. Set to one in https://github.com/plotly/plotly.js/pull/5732/commits/6e6bfa693492867bd9a4c35d2ff75a6755fede8c.